### DISCUSSION TITLE
【QST】Some questions about example 13 two tensor op fusion.

### DISCUSSION THREAD

**Discussion by tianyan01** [NVIDIA/cutlass#1282] [Q&A] [answered]
Hi, 
I'm tring to fuse 
int8 C = gelu(int8 A * int8 B1 * per_channel_scale1 + bias1) * per_element_scale
fp16 D = int8 C * int8 B2 * per_channel_scale2 + bias2
into one kernel, like example 13.
https://github.com/NVIDIA/cutlass/blob/b7508e337938137a699e486d8997646980acfc58/examples/13_two_tensor_op_fusion
But in this example, I found that the problem size N should euqal thread block shape N, the problem size M should be a multiple of kAlignmentA, why? 
And in my code, problem size N is 6144, when I set thread block shape N = 6144, I got an static assertion "Too many predicates". 
How to modify the example 13 to support problem size N  not euqal thread block shape N?
Thanks

**Comment by hwu36** [comment, marked as answer]
if problem size N is larger than threadblock size N, different threadblocks need to communicate with each other, that is why it is not supported.

you need to do splitN to relax this restriction just like what is discussed in the 2nd half of https://www.nvidia.com/en-us/on-demand/session/gtcspring22-s41606/ @lb1244206405
